scene/resources/animation.cpp

```diff
@@ -3799,8 +3905,10 @@ void Animation::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("audio_track_get_key_start_offset", "track_idx", "key_idx"), &Animation::audio_track_get_key_start_offset);
 	ClassDB::bind_method(D_METHOD("audio_track_get_key_end_offset", "track_idx", "key_idx"), &Animation::audio_track_get_key_end_offset);
 
-	ClassDB::bind_method(D_METHOD("bezier_track_set_key_handle_mode", "track_idx", "key_idx", "key_handle_mode", "balanced_value_time_ratio"), &Animation::bezier_track_set_key_handle_mode, DEFVAL(1.0));
+#ifdef TOOLS_ENABLED
+	ClassDB::bind_method(D_METHOD("bezier_track_set_key_handle_mode", "track_idx", "key_idx", "key_handle_mode", "key_handle_set_mode"), &Animation::bezier_track_set_key_handle_mode, DEFVAL(HANDLE_SET_MODE_NONE));
 	ClassDB::bind_method(D_METHOD("bezier_track_get_key_handle_mode", "track_idx", "key_idx"), &Animation::bezier_track_get_key_handle_mode);
+#endif // TOOLS_ENABLED
```

Comment:
We shouldn't bind methods/constants only in `TOOLS_ENABLED` builds. This makes them part of the Core API but only when you run an editor build, which means you could use them while developing your project but it will fail in an exported project. They also appear in the documentation (`--doctool` is a `TOOLS_ENABLED` build), but they shouldn't if they're only meant for editor internal code.

Having `TOOLS_ENABLED` conditional method bindings can also break the Mono bindings in 3.x, I don't know what's the state in 4.0 with the dotnet6 port but either way, this needs a different approach.

Reply:
That should only be necessary for UndoRedo, so I'll try to figure out if it is possible to set it on the Editor side.

Reply:
That method is now bound to the AnimationTrackEditor side.